src/longhorn_rpc_client.c

```diff
@@ -46,14 +45,14 @@ void update_timeout_timer(struct lh_client_conn *conn) {
         its.it_interval.tv_nsec = 0;
 
         // When there are messages on the queue, make timer that expires
-        // request_timeout_period seconds from now.
+        // request_timeout seconds from now.
         if (head_msg != NULL) {
                 if (clock_gettime(CLOCK_MONOTONIC, &its.it_value) < 0) {
                         perror("Fail to get current time");
                         return;
                 }
 
-                its.it_value.tv_sec += request_timeout_period;
+                its.it_value.tv_sec += conn->request_timeout;
```

Comment:
Should we fall back to the default 15 seconds if the value is not provided and it's default is 0?

Reply:
The value will be provided by longhorn-engine, so we don’t need a fallback.
But for safety, I think we can give it a default value as you mentioned. If request_timeout==0 (given by engine and something wrong), then use 15 seconds instead.

Reply:
Just found I set the default value to 15 in [tgt](https://github.com/rancher/tgt/pull/10) part. Then, we don't need any change in this PR. It can be reviewed now.